servers/navigation_server_2d.cpp

```diff
@@ -305,3 +307,170 @@ bool FORWARD_1_C(agent_is_map_changed, RID, p_agent, rid_to_rid);
 void FORWARD_4_C(agent_set_callback, RID, p_agent, Object *, p_receiver, StringName, p_method, Variant, p_udata, rid_to_rid, obj_to_obj, sn_to_sn, var_to_var);
 
 void FORWARD_1_C(free, RID, p_object, rid_to_rid);
+
+Ref<NavigationPathQueryResult2D> NavigationServer2D::query_path(const Ref<NavigationPathQueryParameters2D> &p_query_parameters) const {
+	gd::PathQueryResult _query_result = NavigationServer3D::get_singleton()->_query_path(p_query_parameters->get_parameters());
+
+	Vector<Vector2> path_vec2;
+
+	for (int i = 0; i < _query_result.path.size(); i++) {
+		path_vec2.push_back(Vector2(_query_result.path[i].x, _query_result.path[i].z));
+	}
+
+	Ref<NavigationPathQueryResult2D> query_result(memnew(NavigationPathQueryResult2D));
+
+	query_result->set_path(path_vec2);
+	query_result->set_path_length(_query_result.path_length);
+
+	return query_result;
+}
+
+void NavigationPathQueryParameters2D::set_pathfinding_algorithm(const NavigationPathQueryParameters2D::PathfindingAlgorithm p_pathfinding_algorithm) {
+	switch (p_pathfinding_algorithm) {
+		case PATHFINDING_ALGORITHM_ASTAR: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+		default: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+	}
```

Comment:
Do there need to be two different enums here? Is it possible we can just set one equal to the other?

Reply:
I split that because it is entirely possible that we have different enum values between 2D and 3D and those enums are exposes in docs for users.